examples/dpapi.py

```diff
@@ -503,6 +475,63 @@ def run(self):
                 # Just print the data
                 blob.dump()
 
+        elif self.options.action.upper() == 'CREDHIST':
+            fp = open(options.file, 'rb')
+            data = fp.read()
+            chf = CREDHIST_FILE(data)
+
+            keys = []
+
+            # Handle key options
+            if self.options.key and self.options.sid:
+                key = unhexlify(self.options.key[2:])
+                keys = deriveKeysFromUserkey(self.options.sid, key)
+
+            elif self.options.sid and self.options.key is None:
+                # Do we have a password?
+                if self.options.password is None:
+                    # Nope let's ask it
+                    from getpass import getpass
+                    password = getpass("Password:")
+                else:
+                    password = options.password
+                keys = deriveKeysFromUser(self.options.sid, password)
+
+            else:
+                chf.dump()
+                return
```

Comment:
I think it would be good to add a check in case the CREDHIST file is empty. Additionally, I believe we don't need to request the user's SID (we can obtain it from the CREDHIST file, as many tools with this functionality do). What do you think?
```suggestion
            if not chf.credhist_entries:
                print('The CREDHIST file is empty')
                return

            # Handle key options
            if self.options.key:
                key = unhexlify(self.options.key[2:])
                keys = deriveKeysFromUserkey(chf.credhist_entries_list[0].sid, key)
            # Do we have a password?
            elif self.options.password:
                keys = deriveKeysFromUser(chf.credhist_entries_list[0].sid, options.password)
            else:
                chf.dump()
                print('Cannot decrypt (specify -key or -password)')
                return
```